domains/transfer-agreements/TransferAgreements.API/API/Controllers/TransferAgreementsController.cs

```diff
@@ -170,17 +175,16 @@ public async Task<ActionResult<EditTransferAgreementEndDate>> EditEndDate(Guid i
     }
 
     [ProducesResponseType(typeof(string), 200)]
+    [ProducesResponseType(typeof(string), 500)]
     [HttpPost("wallet-deposit-endpoint")]
     public async Task<ActionResult> CreateWalletDepositEndpoint()
     {
-        var issuer = User.FindIssuerClaim();
-        var audience = User.FindAudienceClaim();
-        var subject = User.FindSubjectGuidClaim();
-        var name = User.FindSubjectNameClaim();
-
-        var base64String = await walletDepositEndpointService.CreateWalletDepositWithToken(new JwtToken(issuer, audience, subject, name));
-
-        return Ok(new { result = base64String });
+        if (AuthenticationHeaderValue.TryParse(httpContextAccessor.HttpContext?.Request.Headers["Authorization"], out var authentication))
+        {
+            var base64String = await walletDepositEndpointService.CreateWalletDepositWithToken(authentication.Parameter);
```

Comment:
The `WalletDepositEndpointService` explicitly adds (hardodes) "Bearer" infront of the token. Why not just pass the AuthenticationHeaderValue object so that `WalletDepositEndpointService` knows both the scheme and the parameter (the token)?

Reply:
It was to refrain from changing more code than necessary. I will update it, and let you know when it is done

Reply:
Updated so it passes the scheme and token now. Authorization still has to be set in the service since we are using grpc.